src/get_window.py

```diff
@@ -37,40 +43,73 @@ def get_list_of_all_windows():
     raise 'Platform %s not supported' % p
 
 def linux_get_list_of_all_windows():
+    '''
+    Added new code for this function, which uses the bash package kdotool for checking all windows and sending them to ret
+    whenever the display server is on wayland. As far as I can tell, it grabs xwayland windows as well. This method should only work for KDE,
+    and I tested it on Manjaro Linux running plasma 6
+    '''
     ret = set()
     ewmh = EWMH()
-    for window in ewmh.getClientList():
-        try:
-            win_pid = ewmh.getWmPid(window)
-        except TypeError:
-            win_pid = False
-        if win_pid:
-            app = psutil.Process(win_pid).name()
-        else:
-            app = 'Unknown'
-        wm_name = window.get_wm_name()
-        if not wm_name:
-            wm_name = window.get_full_property(NET_WM_NAME, 0).value
-        if not wm_name:
-            wm_name = 'class:{}'.format(window.get_wm_class()[0])
-        if isinstance(wm_name, bytes):
-            wm_name = wm_name.decode('utf-8')
-        ret.add((app, wm_name))
+    if is_plasmawayland: #This part of the function uses kdotool to check for all active windows and sends them to ret.
+        for window in list(filter(None, subprocess.check_output(['kdotool', 'search']).decode().split('\n'))):
```

Comment:
Looks like `kdotool` has had some changes, and `search` now requires arguments not being supplied here, so this solution doesn't work in its current form anymore.